akka23-conductr-client-lib/src/main/scala/com/typesafe/conductr/clientlib/akka/BundlesConnector.scala

```diff
@@ -81,56 +165,111 @@ object BundlesConnector {
    * having received events because then startup could fail if there are none i.e. we must always
    * get the initial state of bundle info.
    */
-  private def consumeBundlesEvents(connector: ActorRef, getBundlesEvents: BundlesConnector.GetBundleEvents)(
-    implicit
-    ec: ExecutionContext, mat: ActorMaterializer
-  ): Unit =
-    for {
-      eventStreamResult <- getBundlesEvents()
-    } eventStreamResult match {
-      case EventStreamSuccess(sseSource) =>
-        sseSource
-          .runForeach(connector ! _)
-          .map(done => connector ! ConsumeBundlesEventsOutcome(Success(done)))
-          .recover {
-            case error =>
-              connector ! ConsumeBundlesEventsOutcome(Failure(error))
+  private def consumeBundlesEvents(connector: ActorRef, connection: Connection, apiVersion: ApiVersion.Value)(implicit ec: ExecutionContext, mat: ActorMaterializer): Unit = {
+    import EventStreamUnmarshalling._
+
+    Source
+      .single(bundlesEventsRequest(apiVersion))
+      .via(connection)
+      .alsoTo(
+        Sink.head.mapMaterializedValue(_.foreach { response =>
+          if (response.status.isSuccess())
+            connector ! GetBundles
+        })
+      )
+      .mapAsync(1) { response =>
+        Unmarshal(response.entity.withSizeLimit(Long.MaxValue))
```

Comment:
Just in case if you missed it, there are 2 different versions `BundlesConnector` for Akka 2.3 and Akka 2.4.

The version for Akka 2.4 makes use of `Streams.sseFlow`, while the version for Akka 2.3 doesn't because the Akka SSE version it depends on doesn't have `Streams.sseFlow`.